hazelcast/src/main/java/com/hazelcast/sql/impl/expression/datetime/Formatter.java

```diff
@@ -0,0 +1,1022 @@
+            DD    (ChronoField.DAY_OF_MONTH, 2),
+            D     (ChronoField.DAY_OF_WEEK, 1),
+            ID    (ISO.dayOfWeek(), 1),
+            W     (ChronoField.ALIGNED_WEEK_OF_MONTH, 1),
+            WW    (ChronoField.ALIGNED_WEEK_OF_YEAR, 2),
+            IW    (ISO.weekOfWeekBasedYear(), 2),
+            CC    ((t, l) -> (int) Math.ceil(t.get(ChronoField.YEAR_OF_ERA) / 100f), 2),
+            J     ((t, l) -> t.getLong(JulianFields.JULIAN_DAY), 7),
+            Q     (IsoFields.QUARTER_OF_YEAR, 1),
+            RY    ((t, l) -> toRoman(t.get(ChronoField.YEAR_OF_ERA)), 15),
+            ry    (RY, LOWERCASE),
+            RM    ((t, l) -> toRoman(t.get(ChronoField.MONTH_OF_YEAR)), 4),
+            rm    (RM, LOWERCASE),
+            RD    ((t, l) -> toRoman(t.get(ChronoField.DAY_OF_MONTH)), 6),
+            rd    (RD, LOWERCASE),
+            TZ    ((t, l) -> SIGN.split(TIMEZONE_FORMATTER.withLocale(l).format(t))[0], 3),
+            tz    (TZ, LOWERCASE),
+            TZH   ((t, l) -> ZoneOffset.from(t).getTotalSeconds() / 3600, 2),
+            TZM   ((t, l) -> (ZoneOffset.from(t).getTotalSeconds() % 3600) / 60, 2),
+            OF    ((t, l) -> ZoneOffset.from(t).getId(), 6);
+
+            final BiFunction<Temporal, Locale, Object> query;
+            final int maxLength;
+
+            PatternElement(PatternElement pattern) {
+                this(pattern.query, pattern.maxLength);
+            }
+            PatternElement(TemporalField field, int maxLength) {
+                this((t, l) -> t.get(field), maxLength);
+            }
+            PatternElement(PatternElement pattern, BiFunction<Object, Locale, Object> transform) {
+                this(pattern, transform, pattern.maxLength);
+            }
+            PatternElement(PatternElement pattern, BiFunction<Object, Locale, Object> transform, int maxLength) {
+                this((t, l) -> transform.apply(pattern.query.apply(t, l), l), maxLength);
+            }
+            PatternElement(BiFunction<Temporal, Locale, Object> query, int maxLength) {
+                this.query = query;
+                this.maxLength = maxLength;
+            }
+        }
+
+        enum Ordinal { TH, th }
+
+        class DateTimeGroupProcessor implements GroupProcessor {
+            @Override
+            public void acceptLiteral(String literal) {
+                parts.add(new Literal(literal));
+            }
+
+            @Override
+            public void acceptGroup(String group, Matcher m) {
+                if (group.startsWith("\"")) {
+                    parts.add(new Literal(group));
+                } else {
+                    String prefix = m.group(1).toUpperCase();
+                    String pattern = m.group(2).replace('.', '_').replace(',', '_');
+                    String suffix = m.group(3);
+                    parts.add(new PatternInstance(!prefix.contains("FM"), !prefix.contains("TM"),
```

Comment:
The first parameter of `PatternInstance` is named `fill`. This is confusing, because it's inverted. The flag `FM` means "fill mode". However, if the flag is present, `fill` is false, and vice versa. We should rename the parameter.

Reply:
Postgres documentation describes `FM` pattern as "fill mode (suppress trailing zeroes and padding blanks)." Thus, it contradicts itself by saying "filling operation suppresses the filler zeros/spaces." In our specification, the fill and translate modes are enabled by default and are disabled by introducing `FM` and `TM` patterns respectively. _(We don't do anything differently; we just have a specification that is free from such contradictory statements.)_

Reply:
I also find it confusing, but I don't think it's contradicting. The interpretation depends on what do you mean by "filler". In your understanding it's the blanks and zeroes. In their understanding it's the actual data, so if "fill mode" is on, the data fills the whole output string. Both interpretations are correct IMO, that's why they are confusing, or ambiguous. I also remember the meaning by interpreting FM as "format minimal", which isn't ambiguous.

Since we're following PG (and Oracle), we should again use their interpretation, otherwise we're only adding to the confusion.

The javadoc should also be updated.